### DISCUSSION TITLE
How to match on pattern when POS doesn't apprear

### DISCUSSION THREAD

**Discussion by antonpibm** [explosion/spaCy#10501] [Help: Coding & Implementations]
Hi everyone.

I'm having trouble writing a Matcher rule for validating that a certain POS doesn't appear before some other pattern.
Example: validate that no `PROPN` exist in a sentence before the first `VERB`

Currently, to solve this I have to instantiate 2 Matchers:
1. For validating a `VERB` exist, with pattern - `[{"POS": "VERB"}]`
2. For validating a `PROPN` exist somewhere before the `VERB`, with pattern - `[{"POS": "PROPN"},{"OP": "*"},{"POS": "VERB"}]`

Then in a filtering I need to check `if matcher1 and not matcher2`

Trying to negate the `PROPN` with pattern `[{"IS_SENT_START": True, "POS": {"NOT_IN": ["PROPN"]},"OP": "*"}, {"POS": "VERB"}]` doesn't work, you can test againt: `"I am in Brazil, which is the most beautiful country"`

I guess I can check the matches programmatically, but I wonder if this could be handles via the pattern itself

**Comment by koaning** [comment]
I think the `"NOT_IN"` syntax should work. But when I test against `"I am in Brazil, which is the most beautiful country"` then on the `en_core_web_md` model the token "am" is detected as `AUX` instead of `VERB`. That might explain part of what you're experiencing here. 

```python
import spacy
from spacy.matcher import Matcher
from spacy.displacy import render

nlp = spacy.load("en_core_web_md")

doc = nlp("I am in Brazil, which is the most beautiful country")
render(doc)
```

<img width="1247" alt="image" src="https://user-images.githubusercontent.com/1019791/158754168-3872442e-d367-4957-a556-d57d1b5ce4ef.png">



Could you share a code snippet with an `assert` statement that indicates the expected behavior? 

Also, could you perhaps share a bit more about the actual use-case you're trying to solve?
